subprojects/core/src/main/java/org/gradle/internal/classpath/AccessTrackingSet.java

```diff
@@ -105,14 +129,25 @@ public int hashCode() {
 
     @Override
     public Object[] toArray() {
-        reportAggregatingAccess();
-        return delegate.toArray();
+        // this is basically a reimplementation of the standardToArray that doesn't call this.size()
+        // and avoids double-reporting of the aggregating access.
+        return toArray(new Object[delegate.size()]);
```

Comment:
🚀 `toArray(new Object[0])`?

I actually wonder if this advice from 2016 still holds:

> *Bottom line*: toArray(new T[0]) seems faster, safer, and contractually cleaner, and therefore should be the default choice now. 
https://shipilev.net/blog/2016/arrays-wisdom-ancients/#_conclusion

Reply:
I guess following the wise wouldn't hurt, fixed.